plugins/woocommerce/includes/admin/class-wc-admin-post-types.php

```diff
@@ -897,6 +897,7 @@ private function set_new_price( $product, $price_type ) {
 		}
 
 		$old_price     = (float) $product->{"get_{$price_type}_price"}();
+		$old_price     = ! $old_price ? (float) $product->get_regular_price() : $old_price;
```

Comment:
@imanish003 Good catch! The case where `$old_price` is `0` is wrong, but I'm not sure if should be `10` in the case you mention.
The increase & decrease percentages are calculated from the existing sale amount, not from the product price.
For example:

1. Product price is `100` (regular price)
2. Product is put on sale for `10` (sale price)
3. User increases sale price by 10%. The new sale price will be `11`, not `20` (because 10% of 10 is 1, 10+1=11).

So, following this logic, in the case you mention, the new price should still be `0`, since 10% of `0` is still `0`.
I know it's a bit weird, wdyt?

Reply:
> So, following this logic, in the case you mention, the new price should still be 0, since 10% of 0 is still 0.
> I know it's a bit weird, wdyt?

Ah! I see what you mean. You are right. It should still be 0. 
But in case user increases the sale price by fixed amount i.e. 10, then the new price will be 10. Right?

Reply:
That's correct!
I'll push a fix soon 🙌